tests/e2e_test.go

```diff
@@ -0,0 +1,497 @@
+
+		// Loop over each client on chain A to ensure they all have time to accept the block.
+		// Note: if we did not confirm this here, the next stage could be racy since it assumes every node
+		// has accepted the block.
+		for i, uri := range chainANodeURIs {
+			chainAWSURI := httpToWebsocketURI(uri, blockchainIDA.String())
+			log.Info("Creating ethclient for blockchainA", "wsURI", chainAWSURI)
+			client, err := ethclient.Dial(chainAWSURI)
+			Expect(err).Should(BeNil())
+
+			// Loop until each node has advanced to >= the height of the block that emitted the warp log
+			for {
+				block, err := client.BlockByNumber(ctx, nil)
+				Expect(err).Should(BeNil())
+				if block.NumberU64() >= newHeadA.Number.Uint64() {
+					log.Info("client accepted the block containing SendWarpMessage", "client", i, "height", block.NumberU64())
+					break
+				}
+			}
+		}
+
+		// Get the aggregate signature for the Warp message
+		log.Info("Fetching aggregate signature from the source chain validators")
+		warpClient, err := warpBackend.NewWarpClient(chainANodeURIs[0], blockchainIDA.String())
+		Expect(err).Should(BeNil())
+		signedWarpMessageBytes, err := warpClient.GetAggregateSignature(ctx, unsignedWarpMessageID, params.WarpQuorumDenominator)
+		Expect(err).Should(BeNil())
+
+		// Construct the transaction to send the Warp message to the destination chain
+		log.Info("Constructing transaction for the destination chain")
+		signedMessage, err := avalancheWarp.ParseMessage(signedWarpMessageBytes)
+		Expect(err).Should(BeNil())
+
+		numSigners, err := signedMessage.Signature.NumSigners()
+		Expect(err).Should(BeNil())
+
+		gasLimit, err := teleporter.CalculateReceiveMessageGasLimit(numSigners, teleporterMessage.RequiredGasLimit)
+		Expect(err).Should(BeNil())
+
+		callData, err := teleporter.EVMTeleporterContractABI.Pack("receiveCrossChainMessage", fundedAddress)
+		Expect(err).Should(BeNil())
+
+		baseFee, err := chainBRPCClient.EstimateBaseFee(ctx)
+		Expect(err).Should(BeNil())
+
+		gasTipCap, err := chainBRPCClient.SuggestGasTipCap(ctx)
+		Expect(err).Should(BeNil())
+
+		nonce, err := chainBRPCClient.NonceAt(ctx, fundedAddress, nil)
+		Expect(err).Should(BeNil())
+
+		gasFeeCap := baseFee.Mul(baseFee, big.NewInt(2))
+		gasFeeCap.Add(gasFeeCap, big.NewInt(2500000000))
+		destinationTx := predicateutils.NewPredicateTx(
+			chainBIDInt,
+			nonce,
+			&teleporterContractAddress,
+			gasLimit,
+			gasFeeCap,
+			gasTipCap,
+			big.NewInt(0),
+			callData,
+			types.AccessList{},
+			warp.ContractAddress,
+			signedMessage.Bytes(),
+		)
+
+		// Sign and send the transaction on the destination chain
+		signer := types.LatestSignerForChainID(chainBIDInt)
+		signedTxB, err := types.SignTx(destinationTx, signer, fundedKey)
+		Expect(err).Should(BeNil())
+
+		log.Info("Sending transaction to destination chain")
+		err = chainBRPCClient.SendTransaction(context.Background(), signedTxB)
+		Expect(err).Should(BeNil())
```

Comment:
Similarly, could this be moved to a `broadcastSignedMessage` helper?

Reply:
Added a helper